stacker/context.py

```diff
@@ -129,22 +129,28 @@ def get_stacks(self):
             list: a list of :class:`stacker.stack.Stack` objects
 
         """
-        stacks = []
-        definitions = self._get_stack_definitions()
-        for stack_def in definitions:
-            stack = Stack(
-                definition=stack_def,
-                context=self,
-                mappings=self.mappings,
-                force=stack_def.name in self.force_stacks,
-                locked=stack_def.locked,
-                enabled=stack_def.enabled,
-                protected=stack_def.protected,
-            )
-            stacks.append(stack)
-        return stacks
-
-    def get_stacks_dict(self):
+        if not hasattr(self, "_stacks"):
+            self._stacks = []
+            definitions = self._get_stack_definitions()
+            for stack_def in definitions:
+                stack = Stack(
+                    definition=stack_def,
+                    context=self,
+                    mappings=self.mappings,
+                    force=stack_def.name in self.force_stacks,
+                    locked=stack_def.locked,
+                    enabled=stack_def.enabled,
+                    protected=stack_def.protected,
+                )
+                self._stacks.append(stack)
+        return self._stacks
+
+    def get_stack(self, name):
```

Comment:
Needs docs.

Also, if you're wondering why this doesn't use `get_stacks_dict`, that function indexes by `stack.fqn`, which we don't want. We want to index by the name of the node in the graph.